### DISCUSSION TITLE
Large number of components slows game. Proposal to help eliminate some: More constants

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#9108] [Ideas and suggestions]
- [*] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Large number of logic components cause the game to eventually become slow and lag. A large number of these components in many designs are 'memory' components used purely as constants for mathematical operations.

It would be really nice if components like multiply, division, addition and subtraction could use a constant value for one of the inputs, like exponent component can. This would help eliminate a large number of memory components and associated wires.

Behavior could be: New UI box is added, when blank (default), behavior as now (No signal = No output). when box is filled, that value will be taken as 'default' if no other signal is present on that pin. (It would be nice to be able to provide constants for either pin, as with divide and subtract order matters). This way existing designs will still work exactly as they do now, and sub designers won't notice different behavior unless they fill the box. 

Other components that could benefit: Color component is typically used with 2 or 3 constants and 1 or 2 input wire(s). Some designs use a large number of color components for visual effects. 

Concat is often used to merge strings together for 'status reports'. For that matter, concat having more then 2 inputs would eliminate a lot of concat components in such assemblies, especially since it already supports a 'separator', and often the reports are several lines long, resulting in large meshes of concat and memory components to generate the status report text.

For example, reporting something like 'Reactor condition: 15%' via radio/terminal requires 2 concats and 2 memory components, when it could be handled by 1 3-input concat component with support for constants.

**Comment by Saevon** [comment]
For `Concat Components` we also just need up to 6 inputs I think, which would skip the seperator if any of the gates are NULL
